File: collect_app/src/test/java/org/odk/collect/android/support/BooleanChangeLock.java
package org.odk.collect.android.support;

import org.odk.collect.android.backgroundwork.ChangeLock;

import java.util.function.Function;

public class BooleanChangeLock implements ChangeLock {

    private boolean locked;

    @Override
    public <T> T withLock(Function<Boolean, T> function) {
        return function.apply(!locked);
    }

    public void lock() {
        locked = true;
    }
}

File: testshared/src/main/java/org/odk/collect/testshared/BooleanChangeLock.kt
package org.odk.collect.testshared

import java.util.function.Function
import org.odk.collect.shared.locks.ChangeLock

class BooleanChangeLock : ChangeLock {
    private var locked = false
    override fun <T> withLock(function: Function<Boolean, T>): T {
        return function.apply(!locked)
    }

    fun lock() {
        locked = true
    }
}